uvicorn/supervisors/basereload.py

```diff
@@ -71,6 +64,7 @@ def restart(self):
         self.process.start()
 
     def shutdown(self):
+        self.process.terminate()
```

Comment:
Does this mean my worker can shutdown in middle of transaction happening in, say, Starlette's [backgorund task](https://www.starlette.io/background/)?

Reply:
not sure on this one, 
one thing also it that I miread the doc, it says to terminate before join in case of a pool, we are not in this case....(https://docs.python.org/3/library/multiprocessing.html?highlight=join#multiprocessing.pool.Pool.join)

kind of stuck, dunno how to solve this, this is annoying to get the 20s timeout reached in a container

Reply:
@rafalp I think it wont shutdown in middle of transaction.
Uvicorn handles sigterm, and sends shutdown event to Starlette. 
If it does, it's a Starlette issue (maybe on shutdown Starlette should wait for all background tasks..)